sinks/authentication_type/basicauth/authentication.go

```diff
@@ -56,27 +59,32 @@ func (a *AuthConfig) GetFeatureConfig() []authentication_type.ConfigFeature {
 func (a *AuthConfig) ValidateConfiguration(inputFormat string, input interface{}) error {
 	switch inputFormat {
 	case "object":
+		if _, ok := input.(types.Metadata)[UsernameConfigFeature]; !ok {
+			return errors.Wrap(errors.ErrUsernameNotFound, errors.New("username field was not found"))
+		}
+
+		if _, ok := input.(types.Metadata)[PasswordConfigFeature]; !ok {
+			return errors.Wrap(errors.ErrPasswordNotFound, errors.New("password field was not found"))
+		}
+
 		for key, value := range input.(types.Metadata) {
-			if _, ok := value.(string); !ok {
-				if key == "password" {
-					return errors.Wrap(errors.ErrInvalidPasswordType, errors.New("invalid auth type for field: " + key))
-				}
-				if key == "type" {
-					return errors.Wrap(errors.ErrInvalidAuthType, errors.New("invalid auth type for field: " + key))
-				}
```

Comment:
Is the reason to remove this `type` check because we validate auth type from the authentication config with `GetAuthType()` or something else?

Reply:
I think I re-worked this to match what I did in bearertokenauth.
There was no existence check for required fields and that's one thing I think we should be validating early on.
Now ValidateConfiguration will check for required fields, then it does the type and length checks immediately afterwards.